rotkehlchen/exchanges/ftx.py

```diff
@@ -0,0 +1,514 @@
+            paginate: bool = True,
+    ) -> Dict[str, List[Any]]:
+        ...
+
+    @overload
+    def _api_query(
+            self,
+            endpoint: Literal[
+                'fills',
+                'wallet/deposits',
+                'wallet/withdrawals',
+                'markets',
+                'markets/eth/eur/trades',
+            ],
+            start_time: Optional[Timestamp] = None,
+            end_time: Optional[Timestamp] = None,
+            limit: int = PAGINATION_LIMIT,
+            paginate: bool = True,
+    ) -> List[Dict[str, Any]]:
+        ...
+
+    def _api_query(
+            self,
+            endpoint: str,
+            start_time: Optional[Timestamp] = None,
+            end_time: Optional[Timestamp] = None,
+            limit: int = PAGINATION_LIMIT,
+            paginate: bool = True,
+    ) -> Union[List[Dict[str, Any]], Dict[str, List[Any]]]:
+        """Query FTX endpoint and retrieve all available information if pagination
+        is requested. In case of paginate being set to False only one request is made.
+        Can raise:
+        - RemoteError
+        """
+        if not paginate:
+            final_data_no_pag = self._make_request(
+                endpoint=endpoint,
+                limit=limit,
+                start_time=start_time,
+                end_time=end_time,
+            )
+            return final_data_no_pag
+
+        # If there is pagination we follow the example from the official ftx python example
+        # https://github.com/ftexchange/ftx/blob/master/rest/client.py#L163
+        # In this case the strategy is a while loop leaving fixed the start_time (lower bound)
+        # and decreasing end time (the upper bound) until we fetch all the available information
+        new_end_time = end_time
+        ids = set()
+        final_data: List[Dict[str, Any]] = []
+        while True:
+            step = self._make_request(
+                endpoint=endpoint,
+                limit=limit,
+                start_time=start_time,
+                end_time=new_end_time,
+            )
+
+            assert isinstance(step, list), 'Doing pagination in something that is not a list'
+
+            # remove possible duplicates
+            deduped = [
+                r for r in step if
+                'id' in r.keys() and r['id'] not in ids
+            ]
+            ids |= {r['id'] for r in deduped}
+            final_data.extend(deduped)
+
+            if len(step) == 0:
+                break
+
+            # Avoid deserialization error if there is a bad date
+            times = []
+            for t in step:
+                try:
+                    times.append(deserialize_timestamp_from_date(t['time'], 'iso8601', 'ftx'))
+                except (DeserializationError, KeyError):
+                    continue
+
+            if len(times) != 0:
+                new_end_time = min(times)
+            else:
+                log.error(
```

Comment:
Please change this to self.msg_aggregator.add_error().

The way you have it right now the user will not be able to see that an error happened